packages/renderers/renderer-preact/client.js

```diff
@@ -1,4 +1,4 @@
 import { h, render } from 'preact';
 import StaticHtml from './static-html.js';
 
-export default (element) => (Component, props, children) => render(h(Component, props, h(StaticHtml, { value: children })), element);
+export default (element) => (Component, props, children) => render(h(Component, props, children ? h(StaticHtml, { value: children }) : undefined), element);
```

Comment:
```suggestion
export default (element) => (Component, props, children) => render(h(Component, props, children != null && child != '' ? h(StaticHtml, { value: children }) : undefined), element);
```

This is just for your consideration. Also noting there are similar checks in our render handler.

- Should Astro push `0` or `false` as children? (`children != null`)
- Should Astro not push objects that deliberately stringify to a nullish value or an empty string? (`children != ''`)

Reply:
Good question! I don't think children would ever be `0` or `false` as values but would be the stringified `"0"` or `"false"`. I'll double check.